rpc_jobs/phobos.yml

```diff
@@ -0,0 +1,194 @@
+                sudo git checkout "${env.RPC_BRANCH}"
+                sudo git reset --hard origin/${env.RPC_BRANCH}
+                sudo git submodule update --init
+
+                cd /opt/rpc-maas
+                sudo git fetch --all
+                sudo git stash
+                sudo git reset --hard
+                sudo git checkout ${env.RPC_MAAS_BRANCH}
+                sudo git reset --hard origin/${env.RPC_MAAS_BRANCH}
+              """
+            }
+            stage("Check MaaS Credentials"){
+              // MaaS credentials check must be performed on an internal
+              // slave as it may use the internal identity api which
+              // isn't available from phobos.
+              common.internal_slave(){
+                dir("rpc-eng-ops"){
+                  git branch: env.RPC_ENG_OPS_BRANCH, url: env.RPC_ENG_OPS_REPO, credentialsId: 'rpc-jenkins-svc-github-ssh-key'
+                }
+                sh """#!/bin/bash
+                  scl enable python27 ${WORKSPACE}/rpc-eng-ops/phobos-re-automation/maas_creds_check.sh
+                """
+                // Ensure that the MAAS_TOKEN env var is correct
+                // whether the token was obtained from the service account
+                // or supplied via job param.
+                env.MAAS_TOKEN = readFile file: "MAAS_TOKEN"
+              }
+            }
+            stage("Pre Deployment Verify"){
+              verify()
+            }
+            stage("Pre Deploy Warning"){
+                if (env.PRE_DEPLOY_PAUSE == "true"){
+                  slack_notify("Phobos Deploy starting in 15 minutes.", "warning")
+                  sleep(time: 15, unit: "MINUTES")
+                }
+                slack_notify("Phobos Deploy starting now.", "warning")
+            }
+            stage("Deploy"){
+              if (env.DEPLOY == "true"){
+                sh """#!/bin/bash -xeu
+                  sudo -E rpc-eng-ops/phobos-re-automation/deploy.sh
+                """
+              }
+            }
+            stage("Post Deployment Verification"){
+              // no point verifying twice if we're not running a deploy
+              if (env.DEPLOY == "true"){
+                verify()
+              }
+            }
+            if(env.DEPLOY == "true"){
+              slack_notify("Phobos Deploy Complete", "good")
+            }
+          } // withCredentials
+        } catch (e) {
+          if(env.DEPLOY == "true"){
+            slack_notify("Phobos Deploy Failed", "danger")
+            common.create_jira_issue("RE", "PHOBOS DEPLOY / ${env.BUILD_TAG}")
+          } else {
+            slack_notify("Phobos Verification Failed", "warning")
+          }
+          throw e
+        } finally {
+          sh """#!/bin/bash -xeu
+            # All files in the workspace must be owned by jenkins
+            # so they can be cleaned up by jenkins.
+            sudo chown -R jenkins:jenkins . &>/dev/null
```

Comment:
This seems dangerous.  It may be guaranteed that you're in ${WORKSPACE} here, but I'd do an explicit cd incase something doesn't end up where you think it is.

Reply:
Yes.